Manga reader fixes

therobbiedavis opened the pull request:
# Fixed
- Fixed: Fixed an issue where one some screens images would stack instead of being side by side while using the double setting in the manga reader.
- Fixed: Fixed an issue where the cover and last page would not be positioned correctly when using the double (Mange) setting in the manga reader. (Fixes #2891)
- Fixed: Fixed the issue with review cards provider and score extending outside it's card boundary.
- Fixed: Fixed an issue with double scrollbar when viewing saved smart filters

# Changes
- Changes: Changed how scaling works in the manga reader to be more inline with the expected behavior. Height will scale to viewport height even if it is beyond the image's original dimensions. Width will scale to viewport width even if it is beyond the image's original dimensions. Original will show the image at its intensic dimensions, no scaling is applied. (Potentially fixes #2669)

Width (large resolution):
![image](https://github.com/Kareadita/Kavita/assets/7907265/e5d58390-61cf-4942-ac18-b8fac05dd355)

Width (mobile, portrait):
![image](https://github.com/Kareadita/Kavita/assets/7907265/ed1944c7-6939-4420-9979-84a02dd842d2)

Width (mobile, landscape)
![image](https://github.com/Kareadita/Kavita/assets/7907265/b3d9182c-66f7-4fd2-ac94-23a1a1d4b05e)


Height (large resolution):
![image](https://github.com/Kareadita/Kavita/assets/7907265/0ad35915-6820-4255-a6b3-fa7b41631de8)

Height (mobile, portrait):
![image](https://github.com/Kareadita/Kavita/assets/7907265/b7dd1801-9ed3-4d57-a191-db2357236077)

Height (mobile, landscape)
![image](https://github.com/Kareadita/Kavita/assets/7907265/952760ed-2f93-4dd2-b163-b754d4f5789f)


Original (large resolution):
![image](https://github.com/Kareadita/Kavita/assets/7907265/239af03a-cedd-4fb1-9b34-eec541f4ac82)

Original (mobile, portrait):
![image](https://github.com/Kareadita/Kavita/assets/7907265/0b7915b8-a27a-4643-a70e-82f889ab5ecc)

Original (mobile, landscape):
![image](https://github.com/Kareadita/Kavita/assets/7907265/0c7402e3-96f7-476a-b729-74ee4579dcc1)

therobbiedavis committed:
Fixing stacking issue with double on manga reader

# Fixed
- Fixed: Fixed an issue where one some screens images would stack instead of being side by side while using the double setting in the manga reader.

therobbiedavis committed:
Fixes cover position

Fixes #2891

therobbiedavis committed:
Fixed some scaling inconsistencies

therobbiedavis committed:
Fixing some single reader scaling issues

therobbiedavis committed:
Fixing height on mobile

therobbiedavis committed:
Fixing overflow

majora2007 commented:
Reported back cutoff issues via discord, but overall this is looking MUCH better aligned to how I think the reader should function.

therobbiedavis committed:
Moved to dvh instead of vh

"dynamic viewport size allows the content you design to fit exactly within the viewport, irrespective of the presence of dynamic browser interfaces."
https://developer.mozilla.org/en-US/docs/Web/CSS/length#dynamic

dvh is available for all major browsers: https://caniuse.com/viewport-unit-variants

therobbiedavis committed:
Scaling to smallest viewport dimension for fit-to-height

therobbiedavis committed:
Scaling to smallest viewport dimension for fit-to-height

As requested by Joe

therobbiedavis committed:
fixing wide single position

therobbiedavis committed:
fixing review card

therobbiedavis committed:
fixing smart filter double scrollbar

majora2007 commented:
This will be taken care of in #1426 as I needed to test 2 different PRs at once.

therobbiedavis commented:
@majora2007 I think you have the wrong reference, #1426 is over two years old. I think it should be #2960.
